### PR TITLE
chore(deps): Update GuCDK from 57.0.0 to 59.5.1

### LINKED ISSUE #692: CDK - choose between npm and yarn
The CDK section of this repo seems conflicted about whether to use yarn or npm. We should decide on a consistent approach and use only one of these, for a less confusing developer experience

### npm indicators

- We use a [package-lock.json](https://github.com/guardian/prism/blob/3da10c192449db19c41fce7da296ac587c231e9a/cdk/package-lock.json) instead of a yarn.lock
- [cd/script/ci](https://github.com/guardian/prism/blob/3da10c192449db19c41fce7da296ac587c231e9a/cdk/script/ci) uses only npm commands

### yarn indicators
- [cdk/README.md](https://github.com/guardian/prism/blob/3da10c192449db19c41fce7da296ac587c231e9a/cdk/README.md) refers only to yarn
- All scripts in [cdk/ci/script](https://github.com/guardian/prism/tree/3da10c192449db19c41fce7da296ac587c231e9a/cdk/script) use yarn except `ci`

### PR DESCRIPTION
**Pull request by akash1810**
## What does this change?
Update to the latest version of GuCDK.

Updating GuCDK across 2 major versions includes some breaking changes. These are reflected in the update to the snapshot.

The update was done by regenerating the CDK project via:

```bash
npx @guardian/cdk@latest new --package-manager npm --stage PROD --stage CODE --app prism --stack deploy
```

Consequently a number of files have been deleted as:
- They are redundant (e.g. the `script/build` file, and others like it)
- Or they were configuration that has now been inlined into `package.json` or replaced by shared departmental configuration, e.g. the changes to `tsconfig.json`

Annoyingly some files only include whitespace changes. Therefore its recommended to [review w/out them](https://github.com/guardian/prism/pull/945/files?diff=split&w=1).

Performing the update like this also means we fix #692; we're now using `npm` only.

---
Fixes https://github.com/guardian/prism/issues/692.
Co-authored-by: @SarahJessica

### CODE REVIEW

**Review by jacobwinch — APPROVED** [review]

File: cdk/lib/__snapshots__/prism-ec2-app.test.ts.snap [hunk 1]
```diff
@@ -631,6 +636,7 @@ exports[`The PrismEc2App stack matches the snapshot 1`] = `
         },
         "Port": 443,
         "Protocol": "HTTPS",
+        "SslPolicy": "ELBSecurityPolicy-TLS13-1-2-2021-06",
```
Comment (akash1810) [inline cdk/lib/__snapshots__/prism-ec2-app.test.ts.snap:639]: See https://github.com/guardian/cdk/pull/2306.

File: cdk/lib/__snapshots__/prism-ec2-app.test.ts.snap [hunk 2]
```diff
@@ -93,6 +93,11 @@ exports[`The PrismEc2App stack matches the snapshot 1`] = `
           },
         },
         "MaxSize": "4",
+        "MetricsCollection": [
+          {
+            "Granularity": "1Minute",
+          },
+        ],
```
Comment (akash1810) [inline cdk/lib/__snapshots__/prism-ec2-app.test.ts.snap:100]: See https://github.com/guardian/cdk/pull/2455.

File: cdk/lib/__snapshots__/prism-ec2-app.test.ts.snap [hunk 3]
```diff
@@ -1124,6 +1148,7 @@ exports[`The PrismEc2App stack matches the snapshot 1`] = `
           "InstanceType": "t4g.medium",
           "MetadataOptions": {
             "HttpTokens": "required",
+            "InstanceMetadataTags": "enabled",
```
Comment (akash1810) [inline cdk/lib/__snapshots__/prism-ec2-app.test.ts.snap:1151]: See https://github.com/guardian/cdk/pull/2423.

File: cdk/package.json [hunk 4]
```diff
@@ -1,29 +1,76 @@
...
+    "plugins": [
+      "@typescript-eslint"
+    ],
+    "rules": {
+      "@typescript-eslint/no-inferrable-types": 0,
+      "import/no-namespace": 2,
+      "quotes": [
+        "error",
+        "double"
+      ]
```
Comment (akash1810) [inline cdk/package.json:66]: It looks like the departmental preference is to use single quotes. Overriding that here to keep the diff in this PR as minimal as possible, and thus easier to review. This change will be reverted as it's own PR.

File: cdk/tsconfig.json [hunk 5]
```diff
@@ -1,37 +1,12 @@
 {
-  "ts-node": {
-    "compilerOptions": {
-      "module": "CommonJS"
-    }
-  },
+  "extends": "@guardian/tsconfig/tsconfig.json",
```
Comment (akash1810) [inline cdk/tsconfig.json:2]: Prefer to use the departmental configuration, rather than our own.

File: cdk/package.json (hunk 4 above)
Comment (akash1810) [reply cdk/package.json:66]: https://github.com/guardian/prism/pull/946

**Review by jacobwinch — APPROVED** [review]
